qiskit/circuit/parameterexpression.py

```diff
@@ -35,6 +35,12 @@
 ParameterValueType = Union["ParameterExpression", float]
 
 
+class ParameterTypeError(TypeError):
+    """The type error throw by parameter if we try to cast to number unbind parameter."""
+
+    pass
```

Comment:
I've move this to qiskit.circuit.exceptions, but I'm unable to inherit from QiskitError because in that case, the code doesn't work. Let me know if this is needed. In that case, I will try to understand why this happens.